aclmapping/bank/mapping_test.go

```diff
@@ -0,0 +1,88 @@
+package aclbankmapping
+
+import (
+	"testing"
+
+	pp "github.com/k0kubun/pp/v3"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+
+func cacheTxContext(ctx sdk.Context) (sdk.Context, sdk.CacheMultiStore) {
+	ms := ctx.MultiStore()
+	// TODO: https://github.com/cosmos/cosmos-sdk/issues/2824
+	msCache := ms.CacheMultiStore()
+	return ctx.WithMultiStore(msCache), msCache
+}
+
+
+func TestSendToModuleAccount(t *testing.T) {
+	priv1 := secp256k1.GenPrivKey()
+	addr1 := sdk.AccAddress(priv1.PubKey().Address())
+	priv2 := secp256k1.GenPrivKey()
+	addr2 := sdk.AccAddress(priv2.PubKey().Address())
+	coins     := sdk.Coins{sdk.NewInt64Coin("foocoin", 10)}
+
+	tests := []struct {
+		name          string
+		expectedError error
+		msg           *types.MsgSend
+	}{
+		{
+			name:          "default send",
+			msg:           types.NewMsgSend(addr1, addr2, coins),
+			expectedError: nil,
+		},
+	}
+
+	acc1 := &authtypes.BaseAccount{
+		Address: addr1.String(),
+	}
+	acc2 := &authtypes.BaseAccount{
+		Address: addr2.String(),
+	}
+	accs := authtypes.GenesisAccounts{acc1, acc2}
+	balances := []types.Balance{
+		{
+			Address: addr1.String(),
+			Coins:   coins,
+		},
+		{
+			Address: addr2.String(),
+			Coins:   coins,
+		},
+	}
+
+	app := simapp.SetupWithGenesisAccounts(accs, balances...)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+
+	pp.Default.SetColoringEnabled(false)
+	handler := bank.NewHandler(app.BankKeeper)
+
+	for _, tc := range tests {
+
+		t.Run(tc.name, func(t *testing.T) {
+			handlerCtx, cms := cacheTxContext(ctx)
+			_, err := handler(handlerCtx, tc.msg)
+			depdenencies , _ := MsgSendDependencyGenerator(app.AccessControlKeeper, handlerCtx, tc.msg)
+
+			pp.Println(cms.GetEvents())
+			pp.Println(depdenencies)
```

Comment:
Prints:
```
[]types.Event{
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "\x02\x14GK\xa8\xf4m9\x88\xcf\"\xd9\xc8\x1dw\xe3Wɭ\xa5\xbbafoocoin",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "\n\afoocoin\x12\x0210",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "get",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "\x02\x14GK\xa8\xf4m9\x88\xcf\"\xd9\xc8\x1dw\xe3Wɭ\xa5\xbbafoocoin",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "delete",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "\x02\x14i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\ffoocoin",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "\n\afoocoin\x12\x0210",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "get",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "write",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "\x02\x14i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\ffoocoin",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "\n\afoocoin\x12\x0220",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "set",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "\x01GK\xa8\xf4m9\x88\xcf\"\xd9\xc8\x1dw\xe3Wɭ\xa5\xbba",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "\n /cosmos.auth.v1beta1.BaseAccount\x12/\n-cosmos1ga963ard8xyv7gkeeqwh0c6hexk6twmprt3efh",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "get",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "\x01i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\f",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "\n /cosmos.auth.v1beta1.BaseAccount\x121\n-cosmos1d8r39x7nme84lc3q3n5see3y95s5aasvljkx7a\x18\x01",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "get",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "\x01i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\f",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "\n /cosmos.auth.v1beta1.BaseAccount\x121\n-cosmos1d8r39x7nme84lc3q3n5see3y95s5aasvljkx7a\x18\x01",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "has",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "bank/SendEnabled",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "[]",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "get",
        Index: false,
      },
    },
  },
  types.Event{
    Type:       "resource_access",
    Attributes: []types.EventAttribute{
      types.EventAttribute{
        Key:   "access_type",
        Value: "read",
        Index: false,
      },
      types.EventAttribute{
        Key:   "key",
        Value: "bank/DefaultSendEnabled",
        Index: false,
      },
      types.EventAttribute{
        Key:   "value",
        Value: "true",
        Index: false,
      },
      types.EventAttribute{
        Key:   "operation",
        Value: "get",
        Index: false,
      },
    },
  },
}
```
```
[]accesscontrol.AccessOperation{
  accesscontrol.AccessOperation{
    AccessType:         1,
    ResourceType:       4,
    IdentifierTemplate: "\x02\x14GK\xa8\xf4m9\x88\xcf\"\xd9\xc8\x1dw\xe3Wɭ\xa5\xbba",
  },
  accesscontrol.AccessOperation{
    AccessType:         2,
    ResourceType:       4,
    IdentifierTemplate: "\x02\x14GK\xa8\xf4m9\x88\xcf\"\xd9\xc8\x1dw\xe3Wɭ\xa5\xbba",
  },
  accesscontrol.AccessOperation{
    AccessType:         1,
    ResourceType:       4,
    IdentifierTemplate: "\x02\x14i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\f",
  },
  accesscontrol.AccessOperation{
    AccessType:         2,
    ResourceType:       4,
    IdentifierTemplate: "\x02\x14i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\f",
  },
  accesscontrol.AccessOperation{
    AccessType:         1,
    ResourceType:       14,
    IdentifierTemplate: "\x01i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\f",
  },
  accesscontrol.AccessOperation{
    AccessType:         2,
    ResourceType:       14,
    IdentifierTemplate: "\x01i\xc7\x12\x9b\xd3\xdeO_\xe2 \x8c\xe9\f\xe6$-!N\xf6\f",
  },
  accesscontrol.AccessOperation{
    AccessType:         3,
    ResourceType:       0,
    IdentifierTemplate: "*",
  },
}
```

Reply:
This already points one out about the BankSend being enabled or not not being in our acl ops